arrow-libs/core/arrow-core/src/main/kotlin/arrow/core/computations/iterable.kt

```diff
@@ -0,0 +1,91 @@
+package arrow.core.computations
+
+import arrow.continuations.Effect
+import arrow.continuations.generic.DelimitedScope
+
+/**
+ * Allows to iterate over the [Iterable]<A> receiver and operate on any number of values in order.
+ *
+ * With some special features:
+ * - Not all items need to be consumed.
+ * - If we try to get a value after there are no more items then results in a `null` value.
+ * - At any point we can cancel the iteration resulting in a `null` value.
+ *
+ * Can be used to apply operations to an arbitrary number of items in an [Iterable]<A>:
+ * ```kotlin
```

Comment:
Thanks @pablisco !! Please, it would be awesome if `kotlin:ank` is used for these code snippets. 

And I realized that they are using `partialReduceOrNull` here and `iterateOrNull` in tests. 

Thanks!!

Reply:
Oh, good spot, I renamed the function as it felt a bit long. I'll update 💯